contracts/erc7683/ERC7683OrderDepositor.sol

```diff
@@ -36,134 +41,271 @@ abstract contract ERC7683OrderDepositor is ISettlementContract {
-        }
-
-        if (order.originChainId != block.chainid) {
-            revert WrongChainId();
-        }
-
-        // Extract Across-specific params.
-        (AcrossOrderData memory acrossOrderData, AcrossFillerData memory acrossFillerData) = decode(
-            order.orderData,
-            fillerData
-        );
+        (
+            ResolvedCrossChainOrder memory resolvedOrder,
+            AcrossOrderData memory acrossOrderData,
+            AcrossOriginFillerData memory acrossOriginFillerData
+        ) = _resolveFor(order, fillerData);
 
         // Verify Permit2 signature and pull user funds into this contract
         _processPermit2Order(order, acrossOrderData, signature);
 
         _callDeposit(
-            order.swapper,
+            order.user,
             acrossOrderData.recipient,
             acrossOrderData.inputToken,
             acrossOrderData.outputToken,
             acrossOrderData.inputAmount,
             acrossOrderData.outputAmount,
             acrossOrderData.destinationChainId,
-            acrossFillerData.exclusiveRelayer,
+            acrossOriginFillerData.exclusiveRelayer,
             // Note: simplifying assumption to avoid quote timestamps that cause orders to expire before the deadline.
-            SafeCast.toUint32(order.initiateDeadline - QUOTE_BEFORE_DEADLINE),
+            SafeCast.toUint32(order.openDeadline - QUOTE_BEFORE_DEADLINE),
             order.fillDeadline,
-            getCurrentTime() + acrossOrderData.exclusivityDeadlineOffset,
+            acrossOrderData.exclusivityPeriod,
             acrossOrderData.message
         );
+
+        emit Open(keccak256(resolvedOrder.fillInstructions[0].originData), resolvedOrder);
     }
 
     /**
-     * @notice Constructs a ResolvedOrder from a CrossChainOrder and fillerData.
+     * @notice Opens the order.
+     * @dev Unlike openFor, this method is callable by the user.
+     * @dev This will pull in the user's funds and make the order available to be filled.
      * @param order the ERC7683 compliant order.
-     * @param fillerData Across-specific fillerData.
      */
-    function resolve(CrossChainOrder memory order, bytes memory fillerData)
-        external
+    function open(OnchainCrossChainOrder calldata order) external {
+        (ResolvedCrossChainOrder memory resolvedOrder, AcrossOrderData memory acrossOrderData) = _resolve(order);
+
+        IERC20(acrossOrderData.inputToken).safeTransferFrom(msg.sender, address(this), acrossOrderData.inputAmount);
+
+        _callDeposit(
+            msg.sender,
+            acrossOrderData.recipient,
+            acrossOrderData.inputToken,
+            acrossOrderData.outputToken,
+            acrossOrderData.inputAmount,
+            acrossOrderData.outputAmount,
+            acrossOrderData.destinationChainId,
+            acrossOrderData.exclusiveRelayer,
+            // Note: simplifying assumption to avoid the order type having to bake in the quote timestamp.
```

Comment:
I think this is a fair UX trade off. Not sure why depositor would ever want a different timestamp anyways